Fix up .dot's usage with a label

brettlangdon opened the pull request:
Fixes #72 
A few changes here:

* Fix `.inline-center` to actually center things
  * New changes mean we have to apply `.inline-center` on each child element, not on the parent
* Restructure how we create `.dot`s
  * This also include support for having a label, so no longer uses `.inline-center`

```html
<div class="dot">
  Label:&nbsp;
  <span class="dot__icon"></span>
</div>
```

Or as a stand alone dot with no label

```html
<div class="dot dot__icon"></div>
```

Although, I don't see too much use for them as a stand alone element... but suppose if you needed it.


### Inline center
![image](https://cloud.githubusercontent.com/assets/1320353/21110507/7481456a-c06b-11e6-82e5-f2bdce9da9d0.png)

### Dot
![image](https://cloud.githubusercontent.com/assets/1320353/21110520/82e11180-c06b-11e6-95ce-fc59bd66b894.png)



/cc @underdogio/engineering

brettlangdon commented:
@restlessbit sure this is ok as-is? Would we be better off going with more of an `inline-center` approach here? or ok to make this it's own component?

restlessbit commented on docs/components/dot.md:
````diff
@@ -3,42 +3,52 @@ title: Dot
 
 ```html
-<div class="dot"></div>
+<div class="dot dot__icon"></div>
````
not sure how i missed this (tweaked out on ibuprofen right now), but this is technically not valid BEM syntax.
according to BEM, this should be: 

```html
<div class="dot"><div class="dot__icon"></div></div>
```

brettlangdon replied:
yeah, wasn't entirely sure here, assumed a simpler structure would be better.

restlessbit commented:
Now that I think about, it's probably better to separate `.inline-center` from `.dot`, in case we needed to use `.dot` on its own and don't want or need it to be centered.

If you think `vertical-align: baseline` looks better, you could create a dedicated class just for that and apply it to both the label and dot individually. And if we took that approach we could create a dot with a single element `<div class="dot"></div>` 

![giphy](https://cloud.githubusercontent.com/assets/6979137/21111300/d53f5f38-c06e-11e6-8784-a51080b41b61.gif)

brettlangdon commented:
Ok, fixed up based on comments.

We now have `.inline-center` and `.inline-baseline`.

![image](https://cloud.githubusercontent.com/assets/1320353/21146755/cb98126a-c120-11e6-8f17-6084859a5c9b.png)


Updated `.dot`, made `.dot--small` the default, updated examples.

![image](https://cloud.githubusercontent.com/assets/1320353/21146777/e1df2aae-c120-11e6-93ac-02153061d95a.png)


Here are the changes in context:

![image](https://cloud.githubusercontent.com/assets/1320353/21146784/e756c9b0-c120-11e6-8ecd-68a410da5de0.png)
